pkg/admission/server.go

```diff
@@ -157,7 +167,26 @@ func handleValidation(request admission.AdmissionRequest) (*admission.AdmissionR
 		if err != nil {
 			return nil, err
 		}
+
 		fieldErr = v1a2Validation.ValidateHTTPRoute(&hRoute)
+	case v1a1gatewayGVR:
+		var gateway v1alpha1.Gateway
+		deserializer := codecs.UniversalDeserializer()
+		_, _, err := deserializer.Decode(request.Object.Raw, nil, &gateway)
+		if err != nil {
+			return nil, err
+		}
+
+		fieldErr = v1a1Validation.ValidateGateway(&gateway)
+	case v1a2gatewayGVR:
+		var gateway v1alpha2.Gateway
+		deserializer := codecs.UniversalDeserializer()
```

Comment:
@stevesloka I was about to approve but it seems like #662 was merged in. Could you rebase this on top of it and remove this line and line 174?

Reply:
all rebased!